src/ServiceControl.Transports.SQS.Tests/AmazonSQSQueryTests.cs

```diff
@@ -120,12 +119,13 @@ public async Task RunScenario()
             dictionary.Add(AmazonSQSQuery.AmazonSQSSettings.Region, connectionString.Region);
         }
 
-        query.Initialize(new ReadOnlyDictionary<string, string>(dictionary));
+        query.Initialize(dictionary.AsReadOnly());
 
-        await Task.Delay(TimeSpan.FromMinutes(2), token);
+        // Wait for metrics to become visible, usually takes 20-30 seconds
+        await Task.Delay(TimeSpan.FromMinutes(1), TestContext.CurrentContext.CancellationToken);
```

Comment:
Can't we do call the query in a loop instead (with eg a 1s sleep btw) to avoid waiting a full minute?